<h2 style="text-align: center; background-color: #4d648d; font-family:Arial; color: white; padding: 13px; line-height: 1; border-radius:10px">Initiation Python - Import de fichiers 💻</h2>

# Exercice d'introduction

Faire de la veille sur ces sujets et synthétiser vos recherches dans un support de présentation.

Typologie de données :
 - Structurées vs non structurées : définition, exemples, ...
 - Les différents sources de données. Trouvez les packages en Python qui permet notamment de pouvoir importer/se connecter aux données.  

In [ ]:
# Importer les bibliothèques nécessaires
from google.colab import drive
import pandas as pd

# Monter Google Drive
drive.mount('/content/drive')

# Chemin du fichier dans Google Drive (exemple)
csv_file_path = '/content/drive/MyDrive/nom_du_fichier.csv'  # Remplacez 'nom_du_fichier.csv' par le nom de votre fichier
txt_file_path = '/content/drive/MyDrive/nom_du_fichier.txt'  # Remplacez 'nom_du_fichier.txt' par le nom de votre fichier

# Lire un fichier CSV
df_csv = pd.read_csv(csv_file_path)
print("Contenu du fichier CSV :")
print(df_csv.head())

# Lire un fichier TXT
# Spécifiez le délimiteur si nécessaire, ici on utilise un espace comme exemple
df_txt = pd.read_csv(txt_file_path, delimiter=' ')
print("\nContenu du fichier TXT :")
print(df_txt.head())


<h2 style="text-align: center; background-color: #87c7ec; font-family:Arial; color: white; padding: 13px; line-height: 1; border-radius:10px">MISE EN PRATIQUE</h2>

### Import fichier


Import des fichiers et stockage de manière propre dans des dataframes. Vous risquez d'avoir besoin de fonctions spécifiques, uniquement disponible dans certains packages. N'hésitez pas à regarder sur Google pour vous aider !  
Les 3 fichiers à importer :
> prima.txt  
> mar2012.csv  

In [4]:
# Importer les bibliothèques nécessaires
from google.colab import drive
import pandas as pd

# Monter Google Drive
drive.mount('/content/drive')

# Chemin du fichier dans Google Drive (exemple)
csv_file_path = '/content/drive/MyDrive/Oreegami - Codage/Jour 1/Données/mar2012.csv'  # Remplacez 'nom_du_fichier.csv' par le nom de votre fichier
txt_file_path = '/content/drive/MyDrive/Oreegami - Codage/Jour 1/Données/prima.txt'  # Remplacez 'nom_du_fichier.txt' par le nom de votre fichier

# Lire un fichier CSV
df_csv = pd.read_csv(csv_file_path, encoding = 'latin-1', skiprows=3)
print("Contenu du fichier CSV :")
print(df_csv.head())

# Lire un fichier TXT
# Spécifiez le délimiteur si nécessaire, ici on utilise un espace comme exemple
df_txt = pd.read_csv(txt_file_path, delimiter='\t')
print("\nContenu du fichier TXT :")
print(df_txt.head())


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
Contenu du fichier CSV :
   ANAISH DEPNAISH  INDNATH  ETAMATH  ANAISF DEPNAISF  INDNATF  ETAMATF  AMAR  \
0    1982       75        1        1    1984       99        2        1  2012   
1    1956       69        2        4    1969       99        2        4  2012   
2    1982       99        2        1    1992       99        1        1  2012   
3    1985       99        2        1    1987       84        1        1  2012   
4    1968       99        2        1    1963       99        2        1  2012   

   MMAR  JSEMAINE DEPMAR DEPDOM  TUDOM TUCOM NBENFCOM  
0     1         1     29     99      9   NaN        N  
1     1         3     75     99      9   NaN        N  
2     1         5     34     99      9   NaN        N  
3     1         4     13     99      9   NaN        N  
4     1         6     26     99      9   NaN        N  

Contenu du fichier TXT

<ipython-input-4-2ccf4b89cee6>:13: DtypeWarning: Columns (12) have mixed types. Specify dtype option on import or set low_memory=False.
  df_csv = pd.read_csv(csv_file_path, encoding = 'latin-1', skiprows=3)


#### 2. Afficher les données

In [6]:
df_csv

ANAISH DEPNAISH  INDNATH  ETAMATH  ANAISF DEPNAISF  INDNATF  ETAMATF  \
0         1982       75        1        1    1984       99        2        1   
1         1956       69        2        4    1969       99        2        4   
2         1982       99        2        1    1992       99        1        1   
3         1985       99        2        1    1987       84        1        1   
4         1968       99        2        1    1963       99        2        1   
...        ...      ...      ...      ...     ...      ...      ...      ...   
246118    1948       99        1        4    1955       27        1        1   
246119    1961       99        1        4    1976       99        2        1   
246120    1963       60        1        4    1970      971        1        4   
246121    1973       99        2        1    1985      971        1        1   
246122    1980       99        2        1    1982       90        1        1   

        AMAR  MMAR  JSEMAINE DEPMAR DEPDOM  TUDOM TUCOM NBENFCOM  
0       2012     1         1     29     99      9   NaN        N  
1       2012     1         3     75     99      9   NaN        N  
2       2012     1         5     34     99      9   NaN        N  
3       2012     1         4     13     99      9   NaN        N  
4       2012     1         6     26     99      9   NaN        N  
...      ...   ...       ...    ...    ...    ...   ...      ...  
246118  2012    12         6    978    978      9     P        N  
246119  2012    12         6    978    978      9     P        O  
246120  2012    12         5    978    978      9     P        N  
246121  2012    12         5    978    978      9     P        N  
246122  2012    12         5    978    978      9     P        N  

[246123 rows x 16 columns]

Utiliser les fonctions head() et tail() pour afficher les 5 premières et 5 dernières lignes du dataset

❓ Quelle est la taille du dataset ? Utiliser la fonction shape et interpréter le résultat

In [7]:
df_csv.shape

(246123, 16)

246123 lignes et 16 colonnes

Afficher toutes les colonnes du dataset avec un 'columns'

In [8]:
df_csv.columns

Index(['ANAISH', 'DEPNAISH', 'INDNATH', 'ETAMATH', 'ANAISF', 'DEPNAISF',
       'INDNATF', 'ETAMATF', 'AMAR', 'MMAR', 'JSEMAINE', 'DEPMAR', 'DEPDOM',
       'TUDOM', 'TUCOM', 'NBENFCOM'],
      dtype='object')

Sélectionner uniquement une colonne   
Vous pouvez vous inspirer de cette documentation : [Aide](https://apprendrepython.com/pandas-selectionner-des-lignes-colonnes-dans-un-dataframe-en-indexant/)

In [9]:
df_csv['DEPDOM']

0          99
1          99
2          99
3          99
4          99
         ... 
246118    978
246119    978
246120    978
246121    978
246122    978
Name: DEPDOM, Length: 246123, dtype: object

Regarder le nombre de personnes nées selon les mois, avec MMAR (utiliser la fonction [value_counts](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.value_counts.html) )

In [10]:
df_csv['MMAR'].value_counts()

MMAR
6     42880
9     36071
7     32524
8     31543
5     26350
4     16780
10    14149
12    13446
3      9896
11     9303
2      7151
1      6030
Name: count, dtype: int64

❓ Quelle est la moyenne de MMAR ?

In [11]:
df_csv['MMAR'].mean()

6.9721074422138525

❓ Quelle est la valeur maximale de MMAR ?

In [ ]:
df_csv['MMAR'].max()

Sélectionner uniquement les données pour lesquelles le mois >= 6 et stocker le résultat dans une nouvelle variable

In [12]:
df_csv[df_csv['MMAR']>6]

ANAISH DEPNAISH  INDNATH  ETAMATH  ANAISF DEPNAISF  INDNATF  ETAMATF  \
754       1984       99        2        1    1985      974        1        1   
757       1981       75        2        1    1980       75        1        1   
759       1976       99        2        1    1976       75        1        1   
760       1981       33        1        1    1984       59        1        1   
762       1961       47        1        4    1968       99        2        4   
...        ...      ...      ...      ...     ...      ...      ...      ...   
246118    1948       99        1        4    1955       27        1        1   
246119    1961       99        1        4    1976       99        2        1   
246120    1963       60        1        4    1970      971        1        4   
246121    1973       99        2        1    1985      971        1        1   
246122    1980       99        2        1    1982       90        1        1   

        AMAR  MMAR  JSEMAINE DEPMAR DEPDOM  TUDOM TUCOM NBENFCOM  
754     2012     7         3    974     99      9   NaN        N  
757     2012     7         2     75     99      9   NaN        N  
759     2012     7         4     91     99      9   NaN        N  
760     2012     7         5     44     99      9   NaN        N  
762     2012     7         5     64     99      9   NaN        N  
...      ...   ...       ...    ...    ...    ...   ...      ...  
246118  2012    12         6    978    978      9     P        N  
246119  2012    12         6    978    978      9     P        O  
246120  2012    12         5    978    978      9     P        N  
246121  2012    12         5    978    978      9     P        N  
246122  2012    12         5    978    978      9     P        N  

[137036 rows x 16 columns]

#### 3. Export des données

**Pour écrire des données dans un fichier texte (`.txt`) :**

data = ["Ligne 1", "Ligne 2", "Ligne 3"]

with open("data.txt", "w") as file:  
        file.write("\n".join(data))
    

**Pour écrire des données dans un fichier csv (`.csv`) :**

import pandas as pd

df = pd.DataFrame(data[1:], columns=data[0])  
df.to_csv("data.csv", index=False)


**Pour écrire des données dans un fichier Excel (`.xlsx`) :**

import pandas as pd  
df = pd.DataFrame(data)  
df.to_excel("data.xlsx", index=False)